octoprint_mrbeam/filemanager/__init__.py

```diff
@@ -1 +1,72 @@
 #!/usr/bin/env python
+import os
+
+from octoprint.filemanager import FileManager
+from octoprint.filemanager.destinations import FileDestinations
+from octoprint.filemanager.storage import LocalFileStorage
+
+
+# singleton
+_instance = None
+
+
+def mrbFileManager(plugin):
+    global _instance
+    if _instance is None:
+        _instance = MrbFileManager(plugin)
+    return _instance
+
+
+class MrbFileManager(FileManager):
+    class File:
+        def __init__(self, file_name, content):
+            self.filename = file_name
+            self.content = content
+
+        def save(self, absolute_dest_path):
+            with open(absolute_dest_path, "w") as d:
+                d.write(self.content)
+                d.close()
+
+    def __init__(self, plugin):
+        self._plugin = plugin
+        storage_managers = dict()
+        storage_managers[FileDestinations.LOCAL] = LocalFileStorage(
+            self._plugin._settings.getBaseFolder("uploads")
+        )
+
+        FileManager.__init__(
+            self,
+            self._plugin._analysis_queue,
+            self._plugin._slicing_manager,
+            self._plugin.laserCutterProfileManager,
+            initial_storage_managers=storage_managers,
+        )
+
+    def add_file_to_design_library(self, file_name, content, sanitize_name=False):
```

Comment:
Wouldn't it have been easier to just stick with the regular [add_file](https://docs.octoprint.org/en/1.4.2/modules/filemanager.html#octoprint.filemanager.storage.LocalFileStorage.add_file)? At least for the file name or path - I get it for the content.

Reply:
What do you mean?

Reply:
The file storage class already does some path cleanup. I don't exactly know why it has been split up in a different class, [it just seems to do more of what we want](https://github.com/mrbeam/OctoPrint/blob/ca3ba522305f83063445b0ead08aec6fd7c75c5b/src/octoprint/filemanager/storage.py#L739) :)

Reply:
My double bad : the filemanager uses the storage add_file function. So in this case I presume everything is fine, perhaps we don't need to sanitise the name?

Reply:
The MrbFileManager is only used when we upload something from the code: after downloading a design from the ds, etc. For the normal user upload the file names are sanitized, because this is handled by octoprint, but for the design store case it crashed (with münchen_skyline for example). So I just use this function for this case